compiler/rustc_resolve/src/build_reduced_graph.rs

```diff
@@ -24,41 +24,56 @@ use rustc_expand::expand::AstFragment;
 use rustc_hir::def::{self, *};
 use rustc_hir::def_id::{DefId, LocalDefId, CRATE_DEF_ID};
 use rustc_metadata::creader::LoadedMacro;
-use rustc_middle::metadata::ModChild;
+use rustc_middle::metadata::{AmbiguityModChild, ModChild, Reexport};
 use rustc_middle::{bug, ty};
 use rustc_span::hygiene::{ExpnId, LocalExpnId, MacroKind};
 use rustc_span::symbol::{kw, sym, Ident, Symbol};
 use rustc_span::Span;
+use smallvec::SmallVec;
 
 use std::cell::Cell;
 
 type Res = def::Res<NodeId>;
 
+type IsAmbiguity = bool;
+
 impl<'a, Id: Into<DefId>> ToNameBinding<'a>
-    for (Module<'a>, ty::Visibility<Id>, Span, LocalExpnId)
+    for (Module<'a>, ty::Visibility<Id>, Span, LocalExpnId, IsAmbiguity)
 {
     fn to_name_binding(self, arenas: &'a ResolverArenas<'a>) -> NameBinding<'a> {
-        arenas.alloc_name_binding(NameBindingData {
+        let mut binding = NameBindingData {
             kind: NameBindingKind::Module(self.0),
             ambiguity: None,
             warn_ambiguity: false,
             vis: self.1.to_def_id(),
             span: self.2,
             expansion: self.3,
-        })
+        };
+        if self.4 {
+            binding.ambiguity =
+                Some((arenas.alloc_name_binding(binding.clone()), AmbiguityKind::ExternalCrate));
+        }
+        arenas.alloc_name_binding(binding)
     }
 }
 
-impl<'a, Id: Into<DefId>> ToNameBinding<'a> for (Res, ty::Visibility<Id>, Span, LocalExpnId) {
+impl<'a, Id: Into<DefId>> ToNameBinding<'a>
+    for (Res, ty::Visibility<Id>, Span, LocalExpnId, IsAmbiguity)
+{
     fn to_name_binding(self, arenas: &'a ResolverArenas<'a>) -> NameBinding<'a> {
-        arenas.alloc_name_binding(NameBindingData {
+        let mut binding = NameBindingData {
             kind: NameBindingKind::Res(self.0),
             ambiguity: None,
```

Comment:
```suggestion
            ambiguity: self.4.then(|| (arenas.alloc_name_binding(binding.clone()), AmbiguityKind::ExternalCrate)),
```

Reply:
We can't do this because the ambiguity refers to itself (see ﻿`binding.clone`).